### ISSUE TITLE
Assertion Error During Training

### ISSUE THREAD

**Issue by ykwongaq** [Mr-Bigworth/MMCA#3]
Dear Author,

Thank you for sharing the code and developing this visual grounding model.

I encountered an issue during training, specifically when calculating the loss.

In the `generalized_box_iou` function within the `utils/box_utils.py` file, there is an assertion that ensures `(x2, y2)` should be larger than `(x1, y1)` (lines 77-78):

```python
assert (boxes1[:, 2:] >= boxes1[:, :2]).all()
```

However, I encountered an assertion error at line 77, which seems to be related to the model's predictions. From what I understand, the bounding box layer's output is passed through a sigmoid function, which does not guarantee that `(x2, y2)` is greater than `(x1, y1)`. As a result, this causes the training process to terminate due to the assertion failure.

Could you please advise if there is a proper way to handle this issue? 

If I have misunderstood any part of the process, please kinely correct me.

Thank you for your time!

**Comment by ykwongaq** [comment]
Thank you so much for the prompt reply. Relly appreciate it.

No, I did not encounter the same problem with TransVG, but I will try the `--clip_max_norm`.

Thank you so much

**Comment by Mr-Bigworth** [comment]
> Thank you so much for the prompt reply. Relly appreciate it.
> 
> No, I did not encounter the same problem with TransVG, but I will try the `--clip_max_norm`.
> 
> Thank you so much

You can also try it on all three datasets, and if you still have problems, feel free to contact me.

**Comment by ykwongaq** [comment]
Dear Author,

Thany you for the help. I just found out that `--clip_max_norm` is already by default 0.1.

However, I think problem is the intiialization, because the I do not get this problem anymore when I change the `--detr_model` to `detr-r50-referit.pth`.

Thank you for the help.

However, it raise another problem, it seems that the model artitecture is not the same as there are many missing key in the model weights:
```
Missing keys when loading detr model:
['transformer.encoder.layers.0.text_linear_q.text_linear_w', 'transformer.encoder.layers.0.text_linear_q.text_linear_b', 'transformer.encoder.layers.0.text_linear_q.visual_linear_static_down.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.visual_linear_static_down.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.textual_linear_static_down.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.mm_feat_embed_transform.0.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.mm_feat_embed_transform.0.bias', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.mm_feat_embed_transform.2.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.mm_feat_embed_transform.2.bias',
...
```

I do not encoutner thsi problem in TransVG as well. Please kindly let me know is this issue happened before?

Thank you so much again for the kind help.

**Comment by Mr-Bigworth** [comment]
> Dear Author,
> 
> Thany you for the help. I just found out that `--clip_max_norm` is already by default 0.1.
> 
> However, I think problem is the intiialization, because the I do not get this problem anymore when I change the `--detr_model` to `detr-r50-referit.pth`.
> 
> Thank you for the help.
> 
> However, it raise another problem, it seems that the model artitecture is not the same as there are many missing key in the model weights:
> 
> ```
> Missing keys when loading detr model:
> ['transformer.encoder.layers.0.text_linear_q.text_linear_w', 'transformer.encoder.layers.0.text_linear_q.text_linear_b', 'transformer.encoder.layers.0.text_linear_q.visual_linear_static_down.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.visual_linear_static_down.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.textual_linear_static_down.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.mm_feat_embed_transform.0.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.mm_feat_embed_transform.0.bias', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.mm_feat_embed_transform.2.weight', 'transformer.encoder.layers.0.text_linear_q.gate_fusion.mm_feat_embed_transform.2.bias',
> ...
> ```
> 
> I do not encoutner thsi problem in TransVG as well. Please kindly let me know is this issue happened before?
> 
> Thank you so much again for the kind help.

This log is about our proposed modules, which are not initialized by the pretrained weight. It will not influence the training process.
